Convert flexit_bacnet fireplace mode to climate preset- Rename 'Boost…

magnusoverli opened the pull request:
## Breaking change

⚠️ **The fireplace mode switch is deprecated** (but not removed yet)

The fireplace mode switch entity has been **deprecated and disabled by default**. It will be removed in a future version. Users with existing automations can continue using it, but will receive a repair notification prompting them to migrate to the climate preset.

**For users with existing fireplace switch automations:**

A repair issue will guide you to update your automations:

**Before (deprecated):**
```yaml
service: switch.turn_on
target:
  entity_id: switch.flexit_nordic_fireplace_mode
```

**After (recommended):**
```yaml
service: climate.set_preset_mode
target:
  entity_id: climate.flexit_nordic
data:
  preset_mode: fireplace
```

Additionally, the "Boost" preset has been renamed to "High" to match the device's terminology.

## Proposed change

This PR converts the fireplace mode in the flexit_bacnet integration from a switch entity to a proper climate preset mode, matching how it appears on the physical Flexit device. It also renames the "Boost" preset to "High" to align with device terminology.

**Problem:** The fireplace mode was implemented as a switch entity, but on the physical Flexit device it functions as a ventilation mode just like "Home", "Away", and "High" (previously "Boost"). This inconsistency made it less intuitive for users and didn't match the device's actual interface.

**Solution:**
- Renamed preset: Changed "Boost" → "High" to match device labeling
- Converted fireplace mode: Changed from `switch.flexit_nordic_fireplace_mode` to a climate preset
- Added proper UI elements: Translations and icons for all preset modes
- Technical implementation: Uses `operation_mode` property to detect fireplace state (which correctly reports `OPERATION_MODE_FIREPLACE` when active)
- Preserved functionality: Fireplace mode retains its automatic timeout behavior
- **Smooth migration**: Fireplace switch is deprecated (disabled by default) rather than removed, with repair notifications to guide users
- **Fixed preset switching**: Properly exits fireplace mode when switching to other presets by toggling it off before setting new ventilation mode
- **Fixed deprecation warning**: Repair issue now appears immediately on integration load if switch is enabled

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [x] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to developer documentation pull request:
- Link to frontend pull request:

## Checklist

- [x] I understand the code I am submitting and can explain how it works.
- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.
- [x] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

magnusoverli committed:
Convert flexit_bacnet fireplace mode to climate preset- Rename 'Boost' preset to 'High' to match device terminology- Convert fireplace mode from switch entity to climate preset- Add translations and icons for all preset modes- Use operation_mode for accurate fireplace state detection

magnusoverli commented on homeassistant/components/flexit_bacnet/climate.py:
```diff
@@ -127,20 +131,24 @@ def preset_mode(self) -> str:
         Requires ClimateEntityFeature.PRESET_MODE.
         """
-        return VENTILATION_TO_PRESET_MODE_MAP[self.device.ventilation_mode]
+        return OPERATION_TO_PRESET_MODE_MAP.get(self.device.operation_mode, PRESET_HOME)
```
@copilot open a new pull request to apply changes based on [this feedback](https://github.com/home-assistant/core/pull/155760#discussion_r2487863049)

magnusoverli committed:
Deprecate fireplace mode switch instead of removing it

- Keep fireplace mode switch but disable it by default
- Add deprecation warning when switch is used
- Direct users to use climate preset mode instead
- Provides smoother migration path for existing users

lellky commented:
Good job!

But, I have checked out your code and when running it I locally I found that it does not work properly.

Changing preset to Fireplace using the card for the hvac entity works. But when trying to change back to any other preset does not work.

Also I see no warning or repair issue risen when using the (now) deprecated switch.

magnusoverli committed:
Fix fireplace preset switching and add deprecation warning

- Fix: Switching from fireplace preset to other presets now works correctly
  - When in fireplace mode, toggle it off before setting a new ventilation mode
  - trigger_fireplace_mode() acts as a toggle, so calling it while active turns it off
- Fix: Deprecation warning now appears on integration load if fireplace switch is enabled
  - Previously only showed when switch was actively used
  - Now creates repair issue in async_setup_entry if switch is enabled
- Add missing mock properties (fireplace_ventilation_status, cooker_hood_status) to conftest
- Update snapshots to reflect correct fireplace switch state

magnusoverli commented:
> Good job!
> 
> But, I have checked out your code and when running it I locally I found that it does not work properly.
> 
> Changing preset to Fireplace using the card for the hvac entity works. But when trying to change back to any other preset does not work.
> 
> Also I see no warning or repair issue risen when using the (now) deprecated switch.

Thanks for the feedback! :+1: 

I've tried addressing the issues you brought to light. I have tested with my device locally and changing away from Fireplace-mode now works in the card.

lellky commented:
> Thanks for the feedback! 👍
> 
> I've tried addressing the issues you brought to light. I have tested with my device locally and changing away from Fireplace-mode now works in the card.

I can verify that you latest change works; the repair issue is risen and switching presets works. Great!

lellky reviewed (approved):
Looks good to me!

joostlek commented on homeassistant/components/flexit_bacnet/strings.json:
```diff
@@ -139,5 +153,11 @@
+  },
+  "issues": {
+    "deprecated_fireplace_switch": {
+      "description": "The fireplace mode switch entity `{entity_id}` is deprecated and will be removed in a future version.\n\nFireplace mode has been moved to a climate preset to better match the device interface. Please update your automations to use the climate entity preset instead.\n\n**To migrate:**\n\n1. Open your automation that uses `{entity_id}`\n2. Replace the switch service call with: ...
```
I'd just explain what to do rather than including yaml snippets

magnusoverli committed:
Address reviewer feedback on flexit_bacnet fireplace preset PR

- Fix entity ID lookup using entity registry instead of string replacement
- Remove .get() fallback for preset_mode mapping (all modes covered)
- Remove home/away from translations (predefined presets)
- Remove home/away from icons (predefined presets)
- Simplify deprecation message (remove YAML snippets)
- Remove duplicate deprecation logic from switch.py

joostlek committed:
Merge branch 'dev' into flexit-fireplace-preset

joostlek committed:
Fix

joostlek committed:
Merge branch 'dev' into flexit-fireplace-preset

# Conflicts:
#	tests/components/flexit_bacnet/snapshots/test_switch.ambr

joostlek committed:
Fix
